File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/selectors/NotEqualsSelector.java
package org.evomaster.client.java.controller.mongo.selectors;

import org.evomaster.client.java.controller.mongo.operations.NotEqualsOperation;
import org.evomaster.client.java.controller.mongo.operations.QueryOperation;

import java.util.Objects;

/**
 * { field: { $ne: value } }
 */
public class NotEqualsSelector extends SingleConditionQuerySelector {

    public static final String NOT_EQUALS_OPERATOR = "$ne";

    @Override
    protected QueryOperation parseValue(String fieldName, Object value) {
        Objects.requireNonNull(fieldName);
        return new NotEqualsOperation<>(fieldName, value);
    }

    @Override
    protected String operator() {
        return NOT_EQUALS_OPERATOR;
    }
}

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/selectors/NotEqualsSelector.kt
package org.evomaster.client.java.controller.mongo.selectors

import org.evomaster.client.java.controller.mongo.operations.NotEqualsOperation
import org.evomaster.client.java.controller.mongo.operations.QueryOperation

/**
 * { field: { $ne: value } }
 */
class NotEqualsSelector : SingleConditionQuerySelector() {
    override fun operator(): String {
        return "\$ne"
    }

    override fun parseValue(fieldName: String, value: Any?): QueryOperation = NotEqualsOperation(fieldName,value)
}